### PR TITLE
[pipelineX](fix) Fix pip scanner context bug

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

*** Current BE git commitID: d96278ab21 ***
*** SIGSEGV address not mapped to object (@0x0) received by PID 2068650 (TID 2069026 OR 0x7f757e3c7700) from PID 0; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /root/zfr/doris/be/src/common/signal_handler.h:417
 1# 0x00007F76193FF2B7 in /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/server/libjvm.so
 2# JVM_handle_linux_signal in /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/server/libjvm.so
 3# 0x00007F76193F80AC in /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/server/libjvm.so
 4# 0x00007F761DA820C0 in /lib/x86_64-linux-gnu/libc.so.6
 5# doris::vectorized::Block::swap(doris::vectorized::Block&) at /root/zfr/doris/be/src/vec/core/block.cpp:711
 6# doris::pipeline::ScanOperatorX<doris::pipeline::OlapScanLocalState>::get_block(doris::RuntimeState*, doris::vectorized::Block*, doris::pipeline::SourceState&) at /root/zfr/doris/be/src/pipeline/exec/scan_operator.cpp:1507
 7# doris::pipeline::ScanOperatorX<doris::pipeline::OlapScanLocalState>::get_block_after_projects(doris::RuntimeState*, doris::vectorized::Block*, doris::pipeline::SourceState&) at /root/zfr/doris/be/src/pipeline/exec/scan_operator.h:432
 8# doris::pipeline::PipelineXTask::execute(bool*) at /root/zfr/doris/be/src/pipeline/pipeline_x/pipeline_x_task.cpp:275
 9# doris::pipeline::TaskScheduler::_do_work(unsigned long) at /root/zfr/doris/be/src/pipeline/task_scheduler.cpp:284
10# doris::ThreadPool::dispatch_thread() in /mnt/hdd01/STRESS_ENV/be/lib/doris_be
11# doris::Thread::supervise_thread(void*) at /root/zfr/doris/be/src/util/thread.cpp:499
12# start_thread at /build/glibc-sMfBJT/glibc-2.31/nptl/pthread_create.c:478
13# __clone in /lib/x86_64-linux-gnu/libc.so.6 

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

File: be/src/vec/exec/scan/pip_scanner_context.h [hunk 1]
```diff
@@ -297,5 +281,121 @@ class PipScannerContext : public vectorized::ScannerContext {
...
+                       int64_t max_bytes_in_blocks_queue,
+                       std::shared_ptr<pipeline::ScanDependency> dependency,
+                       std::shared_ptr<pipeline::Dependency> finish_dependency)
+            : PipScannerContext(state, local_state, output_tuple_desc, output_row_descriptor,
+                                scanners, limit_, max_bytes_in_blocks_queue, dependency,
+                                finish_dependency) {}
+    Status get_block_from_queue(RuntimeState* state, vectorized::BlockUPtr* block, bool* eos,
+                                int id, bool wait = false) override {
+        {
+            std::unique_lock l(_transfer_lock);
```
Comment (yiguolei) [inline be/src/vec/exec/scan/pip_scanner_context.h:301]: line301 and line313 do not need lock 2 times.
merge these 2 code blocks

**Comment by Gabriel39** [comment]
run buildall

**Review by BiteTheDDDDt — APPROVED** [review]

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**